build.sbt

```diff
@@ -185,6 +191,44 @@ lazy val httpTests = project("akka-http-tests")
   .addAkkaModuleDependency("akka-actor-typed", "provided")
   .addAkkaModuleDependency("akka-multi-node-testkit", "test")
 
+lazy val copyDeps = taskKey[Unit]("Copies all dependencies in a special directory")
+
+lazy val osgiTest = taskKey[Unit]("Executes OSGi integration tests")
+
+lazy val httpOsgiTests = project("akka-http-osgi-tests")
+  .settings(Dependencies.httpOsgiTests)
+  .settings(
+    fork := true,
+    copyDeps := {
+      val httpFile = (http / Compile / OsgiKeys.bundle).value
+      val httpCoreFile = (httpCore / Compile / OsgiKeys.bundle).value
+      val parsingFile = (parsing / Compile / OsgiKeys.bundle).value
+      val dependencies = (Compile / fullClasspath).value.files ++
+        Seq(httpFile, httpCoreFile, parsingFile)
+      val depsTarget = new File(target.value, "dependencies")
+      IO.createDirectory(depsTarget)
+      IO.copy(dependencies.filter(_.isFile)
+        .filterNot(_.getName contains "akka-http-osgi-tests")
+        .map(f => (f, new File(depsTarget, f.getName))))
+
+      val testDependencies = (Test / dependencyClasspath).value.files
+      IO.copy(testDependencies
+        .filter(f => f.getName.contains("scalatest") || f.getName.contains("scalactic") ||
+          f.getName.contains("scala-reflect") || f.getName.contains("scala-xml") ||
+          f.getName.contains("scala-compiler"))
+        .map(f => (f, new File(depsTarget, f.getName))))
+    },
+    osgiTest in Test := Def.sequential(
+      copyDeps,
+      test in Test
+    ).value
```

Comment:
I wonder if we could add `copyDeps` as a precondition of `test in Test` for the `akka-http-osgi-tests` project, that way we might not need to introduce the separate `osgiTest` task?